streamingccl: add more metrics in stream ingestion

gh-casper opened the pull request:
Add more metrics on the stream ingestion including latency related
metrics and etc.

Release note (ui change): add stream ingestion metrics

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/79714)
<!-- Reviewable:end -->

miretskiy commented on pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go:
```diff
@@ -592,20 +595,31 @@ func (sip *streamIngestionProcessor) bufferCheckpoint(event partitionEvent) erro
 	if err := sip.batcher.Flush(sip.Ctx); err != nil {
 		return nil, errors.Wrap(err, "flushing")
 	}
+	sip.metrics.FlushHistNanos.RecordValue(timeutil.Since(preFlushTime).Nanoseconds())
```
wrap lines please?

miretskiy commented on pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go:
```diff
@@ -592,20 +595,31 @@ func (sip *streamIngestionProcessor) bufferCheckpoint(event partitionEvent) erro
 		totalSize += len(kv.Key.Key) + len(kv.Value)
 	}
 
+	preFlushTime := timeutil.Now()
```
looks like this won't record if flush returned an error?
do you think something like this would work:
```
defer func() {
  start := timeutil.Now()
  return func() {
   sip.metrics.FlushHistNanos.RecordValue(
     timeutil.Since(start).Nanoseconds())
  }
}()()

gh-casper committed:
streamingccl: add more metrics in stream ingestion

Add more metrics on the stream ingestion including latency related
metrics and etc.

Release note (ui change): add stream ingestion metrics

gh-casper reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/79714)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @miretskiy and @stevendanna)

___
*[`pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go` line 617 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/79714#-N-BTjU-HbUOElOaLCUU-r1-617:-N4010h-45GRlDtem-qg:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/3c88b9ff258215b948690729badd9ffe6161fbdb/pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go#L617)):*
<details><summary><i>Previously, miretskiy (Yevgeniy Miretskiy) wrote…</i></summary><blockquote>

looks like this won't record if flush returned an error?
do you think something like this would work:
```
defer func() {
  start := timeutil.Now()
  return func() {
   sip.metrics.FlushHistNanos.RecordValue(
     timeutil.Since(start).Nanoseconds())
  }
}()()
```

</blockquote></details>

Done.
___
*[`pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go` line 621 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/79714#-N-BTjU-HbUOElOaLCUU-r1-621:-N4011Os18Ty_MEXn5KN:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/3c88b9ff258215b948690729badd9ffe6161fbdb/pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go#L621)):*
<details><summary><i>Previously, miretskiy (Yevgeniy Miretskiy) wrote…</i></summary><blockquote>

wrap lines please?
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

gh-casper commented:
bors r+

gh-casper commented:
bors r+
